Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -463,6 +463,10 @@ void ComponentInfo::setDetectorInfo(DetectorInfo *detectorInfo) {
     throw std::invalid_argument("ComponentInfo must have detector indices "
                                 "input of same size as size of DetectorInfo");
   }
+  if (detectorInfo && (m_scanIntervals != detectorInfo->scanIntervals())) {
+    throw std::invalid_argument(
+        "ComponentInfo does not have the same scanIntervals as DetectorInfo");
```

Comment:
I think this is confirming what I mention above: Here you check for mismatching scan intervals, nevertheless the API for changing them later is available.